### PR TITLE
scheduling: add a `rename` callback to scheduling_group_key_config

### PR DESCRIPTION
**Pull request by michoecho**
Per-scheduling-group values might want to know the name of their scheduling group, e.g. if they hold some metrics for the group. But scheduling groups can be renamed, and currently there is no way to propagate that info to the values.

This patch allows the user of per-scheduling-group values to pass a callback called when the scheduling group is renamed. As with its sibling callbacks (`constructor` and `destructor`) it is called in the relevant scheduling group, so the value can learn the new name of its group using `current_scheduling_group()`.

### CODE REVIEW

**Comment by michoecho** [comment]
Note: I'm silently assuming that a renamed scheduling group stays alive until the rename call finishes, even though such precondition isn't documented. If this precondition isn't fulfilled, undefined behavior will occur. I leave it to the maintainers to decide whether this is an acceptable assumption.

A semaphore for mutual exclusion of delete/rename could be added to enforce the lifetimes.

**Review by eliransin — APPROVED** [review]
LGTM :slightly_smiling_face:

**Comment by avikivity** [comment]
> Note: I'm silently assuming that a renamed scheduling group stays alive until the rename call finishes, even though such precondition isn't documented. If this precondition isn't fulfilled, undefined behavior will occur. I leave it to the maintainers to decide whether this is an acceptable assumption.


It's implicit, if you pass a scheduling_group parameter to a function, then you're expected to keep it alive until the returned future is ready. Sort of like if you pass a reference to something to a function, it's expected to remain stable until the function returns.

**Comment by michoecho** [comment]
v2: added a unit test as requested.

Please re-review and merge.

**Comment by michoecho** [comment]
@scylladb/seastar-maint Ping.

**Pull request merged**
